allennlp/commands/evaluate.py

```diff
@@ -112,6 +145,13 @@ def evaluate_from_args(args: argparse.Namespace) -> Dict[str, Any]:
     logger.info("Reading evaluation data from %s", evaluation_data_path)
     instances = dataset_reader.read(evaluation_data_path)
 
+    embedding_sources: Dict[str, str] = (json.loads(args.embedding_sources_mapping)
+                                         if args.embedding_sources_mapping else {})
+    if args.extend_vocab:
+        logger.info("Vocabulary is being extended with test instances.")
+        model.vocab.extend_from_instances(Params({}), instances=instances)
+        model.extend_embedder_vocab(model.vocab, embedding_sources)
```

Comment:
Doesn't belong to this PR, but I feel, `extend_embedder_vocab` API should have been just

```python
extend_embedder_vocab(self, embedding_sources_mapping: Dict[str, str] = None):
```
Passing `extended_vocab` isn't necessary and potentially unsafe. We only should allow vocab that's for sure extended from original model.vocab. Allowing to pass arbitrary `extended_vocab` can put user in weird issues if one tries to use it API as standalone and doesn't know that extended_vocab must be extended from original model.vocab.

Since vocab extension is anyway happening in-place, following should work just fine:

```python
model.vocab.extend_from_instances(instances)
model.extend_embedder_vocab(embedding_sources_mapping)
```

And is cleaner+safer than current:
```python
model.vocab.extend_from_instances(instances)
model.extend_embedder_vocab(model.vocab, embedding_sources_mapping)
```

Reply:
I agree with this.  This whole embedding extension hasn't been released yet, so it's fine to simplify this in a subsequent PR.